src/datapoints-graph/export-csv/export-csv.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component } from '@angular/core';
+import {
+  CommonModule,
+  CoreModule,
+  FormsModule,
+  ModalModule,
+} from '@c8y/ngx-components';
+import { BsModalRef } from 'ngx-bootstrap/modal';
+import { CSVExportData } from './export-csv.model';
+
+@Component({
+  selector: 'c8y-export-csv',
+  templateUrl: './export-csv.component.html',
+  standalone: true,
+  imports: [CommonModule, FormsModule, ModalModule, CoreModule],
+})
+export class ExportCsvComponent {
+  readonly defaultSeparator = ',';
+  readonly defaultFilename = 'export';
+  form = {
+    fileName: this.defaultFilename,
+    separator: this.defaultSeparator,
+  };
+
+  csvData: CSVExportData[];
+
+  isLoading = false;
+
+  constructor(public bsModalRef: BsModalRef) {}
+
+  export() {
+    this.isLoading = true;
+    import('file-saver').then((fileSaver) => {
+      for (const series of this.csvData) {
+        const dataSet = series.data
+          .map((value: string[]) => value.join(this.form.separator))
+          .join('\n');
+        const blob = new Blob([dataSet], { type: 'text/csv;charset=utf-8' });
+        fileSaver.default.saveAs(blob, `${this.form.fileName}${series.id}.csv`);
+      }
+
+      this.isLoading = false;
+      this.bsModalRef.hide();
+    });
```

Comment:
```suggestion
import { saveAs } from 'file-saver';
...
    this.isLoading = true;
    for (const series of this.csvData) {
      const dataSet = series.data
        .map((value: string[]) => value.join(this.form.separator))
        .join('\n');
      const blob = new Blob([dataSet], { type: 'text/csv;charset=utf-8' });
      saveAs(blob, `${this.form.fileName}${series.id}.csv`);
    }

    this.isLoading = false;
    this.bsModalRef.hide();
```

Reply:
why using import in method? package file-saver is rather lightweightm, does it need lazy loading? also, after adding types, it's usage is more predictable (e.g. I immediately know that it is synchronous, but I was not sure before)